debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -380,7 +399,12 @@ private Schema makeUpdatedSchema(List<FieldReference> additionalFields, Schema s
 
         // Update the schema with the new fields
         for (FieldReference fieldReference : additionalFields) {
-            builder = updateSchema(fieldReference, builder, originalRecordValue.schema());
+            try {
+                builder = updateSchema(fieldReference, builder, originalRecordValue.schema());
+            }
+            catch (IllegalArgumentException e) {
```

Comment:
Please check `io.debezium.transforms.ExtractNewRecordState.FieldReference.isInSchema(Schema)`.
Could the same approach be used instead of relying on exception handling?

Reply:
@jpechane refactored the code to avoid exception handling.